### PR TITLE
Segregate the Repository reader interface

### PR DESCRIPTION
**Pull request by collheesi**
Introduce `RepositoryReader` and use it everywhere possible (mostly through `RepositoryLocator#snapshotAt`)

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java [hunk 1]
```diff
@@ -86,7 +87,7 @@ public EthModule(
 
     public Map<String, Object> bridgeState() throws IOException, BlockStoreException {
         Block bestBlock = blockchain.getBestBlock();
-        Repository repository = repositoryLocator.snapshotAt(bestBlock.getHeader()).startTracking();
+        Repository repository = repositoryLocator.startTrackingAt(bestBlock.getHeader());
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java]: I think that through this refactor an `up-to-grab` contribution may have rise. It makes no sense to need the whole repository for obtaining the `bridgeState`, a _reader_ should be safer and enough

Comment (collheesi) [reply rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java]: Yes, I tried but the `Bridge` components both read and write. That's the ISP principle right there.

**Comment by collheesi** [comment]
@diega thanks, just fixed those issues! For the bridge one I'll let the `fed` guys know and let them add it to their backlog if they desire.

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java (hunk 1 above)
Comment (diega) [reply rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java]: Absolutely, that's why I think it's a pretty safe place for an external contributor to dig in

**Comment by collheesi** [comment]
Closed in favor of #907 because of the pipeline rejecting PRs from forks.

**Pull request closed without merging**